### ISSUE TITLE
Behavior with `missing`

### ISSUE THREAD

**Issue by jkrumbiegel** [SymbolicML/DynamicQuantities.jl#68]
When trying to integrate DynamicQuantities support into an existing code base, I noticed that one difficulty came from its different behavior regarding `missing`.

```julia
julia> missing * DynamicQuantities.us"hr"
(missing) hr

julia> missing * u"hr"
missing
```

The lower version is Unitful. Because `missing`s are wrapped within `Quantity` in DynamicQuantities, a whole host of missing-handling functions doesn't work and quantities have to be unwrapped manually. As the simplest example, `ismissing(missing * DynamicQuantities.us"hr") === false`. What was the rationale for this behavior, could it still be changed?

**Comment by MilesCranmer** [comment]
Hi @jkrumbiegel,

Thanks for this note! I would consider this a bug because supporting missing-handling functions should be important, but hasn't really been considered in depth. So the current behavior should be changed.

Do you think you could create a suite of unittests for the behavior you are looking for? (Maybe in a PR.) Then I can add the missing functions to make them pass.

Cheers,
Miles

**Comment by MilesCranmer** [comment]
Actually I think once #49 merges, this will already be working, because then `AbstractQuantity <: Number` and therefore the built-in functions for `missing` would already propagate!

**Comment by MilesCranmer** [comment]
Cool!

> The remaining problems were all due to `zero` not being defined on `Type{Quantity}` but that's expected.

Yeah, unfortunately this is a fundamental difference between type-based units and value-based units. The solution is to switch to using `zero(::Quantity)` so it can see both the values and the types – this will work for both Unitful and DynamicQuantities.

**Comment by gaurav-arya** [comment]
> Yeah, unfortunately this is a fundamental difference between type-based units and value-based units. The solution is to switch to using zero(::Quantity) so it can see both the values and the types – this will work for both Unitful and DynamicQuantities.

A nice analogy could be arrays/matrices also have the same behavior, requiring `zero` of the value since the size is not part of the type. This also means we can at least expect any functions that have been written to support array inputs (e.g. `sum`) to already get this right 🙂

**Comment by MilesCranmer** [comment]
Closed with v0.8!
